src/libostree/ostree-deployment.h

```diff
@@ -107,4 +107,12 @@ const char *ostree_deployment_unlocked_state_to_string (OstreeDeploymentUnlocked
 _OSTREE_PUBLIC
 OstreeDeploymentUnlockedState ostree_deployment_get_unlocked (OstreeDeployment *self);
 
+_OSTREE_PUBLIC
+gboolean ostree_deployment_set_ext_metadata (OstreeDeployment *self, const char *metadata_key,
```

Comment:
New API needs to be added to `libostree-devel.sym`; there's some ceremony to do that.  See the comments around that and `git log`.